GitUI/CommandsDialogs/BrowseDialog/FormUpdates.cs

```diff
@@ -145,11 +148,27 @@ private void linkChangeLog_LinkClicked(object sender, LinkLabelLinkClickedEventA
             Process.Start("https://github.com/gitextensions/gitextensions/blob/master/GitUI/Resources/ChangeLog.md");
         }
 
-        private void btnDownloadNow_Click(object sender, EventArgs e)
+        private async void btnDownloadNow_Click(object sender, EventArgs e)
         {
             try
             {
-                Process.Start(UpdateUrl);
+                progressBar1.Visible = true;
+                btnDownloadNow.Enabled = false;
+                UpdateLabel.Text = _downloadingUpdate.Text;
+
+                WebClient webClient = new WebClient();
+                string fileName = UpdateUrl.Substring(UpdateUrl.LastIndexOf("/") + 1);
+                await webClient.DownloadFileTaskAsync(new Uri(UpdateUrl), Environment.GetEnvironmentVariable("TEMP") + "\\" + fileName);
+
+                Process process = new Process();
+                process.StartInfo.UseShellExecute = false;
+                process.StartInfo.FileName = "msiexec.exe";
+                process.StartInfo.Arguments = string.Format("/i \"{0}\\{1}\" /qb LAUNCH=1", Environment.GetEnvironmentVariable("TEMP"), fileName);
+                process.Start();
+
+                progressBar1.Visible = false;
+                Close();
+                Application.Exit();
```

Comment:
I think we need to use JTF to prevent blocking the UI
```
                progressBar1.Visible = true;
                btnDownloadNow.Enabled = false;
                UpdateLabel.Text = _downloadingUpdate.Text;

                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                {
                    await TaskScheduler.Default;

                    // download and execute

                    await this.SwitchToMainThreadAsync();
                    progressBar1.Visible = false;
                    Close();
                    Application.Exit();
                });
```

Reply:
There isn't any code here that blocks the UI. The WebClient is called using async keyword meaning its run on a seperate thread from the UI thread.

Launching the MSI doesn't block and then we close GitExtensions after we launch the MSI. If it blocked then GitExtensions would still be running during the MSI process which I haven't seen to be the case in my testing.